### DISCUSSION TITLE
Cant use template hook inside an extension

### DISCUSSION THREAD

**Discussion by NigmaX** [directus/directus#17966] [Archive]
Hi, im trying to create a custom interface that get an html string and renders it.

Im trying to use the template hook in vue, but it doesn’t seem to work.

```
export default{
Template: “<div>example template</div>”
}
```

also trying to use the render function but it also doesn’t seem to work properly 

```
render:()=>{
 let comp = {
  template: “<div>example template</div>”
 }
 return h(comp)
}
```

Both of the examples above does work on a scaffolded vue application.

any thoughts?
Thanks in advance.

**Comment by nickrum** [comment]
AFAIK the version of Vue that is bundled with the Directus App doesn't include the template compiler which is required to be able to use templates, so you have to either use single file components (which is the faster and recommended way when a bundler is used anyway) or render functions.
